### PR TITLE
Better Indent / De-Indent handling with indent buttons

### PR DESCRIPTION
**Pull request by harshad1**
I threw this together as an stop-gap 'solution' to #844 . But it also implements some useful improvements to indent handling.

This PR adds two completely new indent and de-indent buttons to the end of the markdown toolbar. These buttons trigger an indent routine which is similar to `runMarkdownRegularPrefixAction`. This routine has the following differences from `indentCurrentLine`

1. Indent size (number of spaces) is derived from `_appSettings.getTabWidth()`
2. Indent of multiple lines is handled correctly (using a selection)
3. The cursor position relative to text is maintained upon indent / de-indent

The functionality of the indent and de-indent options in the keyboard menu has been left unchanged.

### CODE REVIEW

**Comment by gsantner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5252618&bid=17212120)

File: app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextActions.java [hunk 1]
```diff
@@ -71,6 +71,8 @@ public boolean runAction(String action, boolean modLongClick, String anotherArg)
             {R.string.tmaid_markdown_horizontal_line, R.drawable.ic_more_horiz_black_24dp, R.string.horizontal_line},
             {R.string.tmaid_markdown_strikeout, R.drawable.ic_format_strikethrough_black_24dp, R.string.strikeout},
             {R.string.tmaid_common_accordion, R.drawable.ic_arrow_drop_down_black_24dp, R.string.accordion},
+            {R.string.tmaid_common_deindent, R.drawable.ic_format_indent_decrease_black_24dp, R.string.deindent},
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextActions.java]: As said, please don't add buttons to the bar for now, instead pursue custom order. Remove the addition and only call the `runIndentLines` in the SpecialKeys dialog.

**Comment by harshad1** [comment]
It will take me a while to make these changes. Please merge my other PR (https://github.com/gsantner/markor/pull/883) if it is acceptable. I will work on fixing this PR later in the week.

**Comment by gsantner** [comment]
OK, btw this should be a matter of deleting two lines and pushing :D

**Comment by harshad1** [comment]
I have deleted the lines.

There will be a conflict between my two PRs. I will rebase after you have merged one of them into master.

**Comment by gsantner** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/net/gsantner/markor/ui/hleditor/TextActions.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5252618&bid=17212120)

**Pull request merged**